tests/pipeline/test_pipeline.py

```diff
@@ -791,6 +791,56 @@ def test_only_nodes_unknown(self, pipeline_list_with_lists, target_node_names):
         with pytest.raises(ValueError, match=pattern):
             full.only_nodes(*target_node_names)
 
+    @pytest.mark.parametrize(
+        "namespaced_node", ["node1", "node2", "node3", "node4", "node5", "node6"]
+    )
+    def test_only_nodes_with_namespacing(
+        self, pipeline_with_namespaces, namespaced_node
+    ):
+        # Tests that error message will supply correct namespaces.
+        # Example of expected error:
+        # Pipeline does not contain nodes named ['node1']. Did you mean: ['katie.node1']?
+        pattern = (
+            rf"Pipeline does not contain nodes named \['{namespaced_node}'\]\. "
+            rf"Did you mean: \['.*\.{namespaced_node}'\]\?"
+        )
+        full = pipeline_with_namespaces
+        with pytest.raises(ValueError, match=pattern):
+            full.only_nodes(namespaced_node)
+
+    @pytest.mark.parametrize(
+        "namespaced_nodes", [("node1", "node2"), ("node3", "node4"), ("node5", "node6")]
+    )
+    def test_only_nodes_with_namespacing_multiple_args(
+        self, pipeline_with_namespaces, namespaced_nodes
+    ):
+        # Tests that error message will contain suggestions for all provided arguments.
+        # Example of expected error message:
+        # "Pipeline does not contain nodes named ['node1', 'node2'].
+        # Did you mean: ['katie.node1', 'lisa.node2']?"
+        pattern = rf"('.*\.{namespaced_nodes[0]}')+.*('.*\.{namespaced_nodes[1]}')+"
```

Comment:
Are `Pipeline does not contain nodes named ...` missing in the `pattern?`

Reply:
The test here is just to make sure that suggestions for both arguments appear in the string. This is because the `test_only_nodes_with_pipeline` tests for the full error and the suggestions cannot appear without the surrounding text.